datafusion/physical-plan/src/joins/hash_join.rs

```diff
@@ -1321,7 +1335,11 @@ impl HashJoinStream {
             None => {
                 self.state = HashJoinStreamState::ExhaustedProbeSide;
             }
-            Some(Ok(batch)) => {
+            Some(Ok(mut batch)) => {
+                if let Some(dynamic_filters) = &mut self.dynamic_filter_info {
+                    batch = dynamic_filters.filter_batch(&batch)?;
```

Comment:
I add the filter logic here, don't know if it is a good choice compare to adding a FilterExec down to scan.